perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -34,34 +34,10 @@
 	 * @throws InternalErrorException when problem
 	 */
 	protected static void init(PerunSession sess) throws InternalErrorException {
-
 		log.trace("Initializing AuthzResolver for [{}]", sess.getPerunPrincipal());
 
-		//Prepare service roles like engine, service, registrar, perunAdmin etc.
-		prepareServiceRoles(sess);
-
-		if (!sess.getPerunPrincipal().getRoles().isEmpty()) {
-			// We have some of the service principal, so we can quit
-			sess.getPerunPrincipal().setAuthzInitialized(true);
-			return;
-		}
-
-		// Prepare first users rights on all subgroups of groups where user is GroupAdmin and add them to AuthzRoles of the user
-		AuthzRoles authzRoles = addAllSubgroupsToAuthzRoles(sess, authzResolverImpl.getRoles(sess.getPerunPrincipal().getUser()));
-
-		// Load all user's roles with all possible subgroups
-		sess.getPerunPrincipal().setRoles(authzRoles);
-
-		// Add self role for the user
-		if (sess.getPerunPrincipal().getUser() != null) {
-			sess.getPerunPrincipal().getRoles().putAuthzRole(Role.SELF, sess.getPerunPrincipal().getUser());
-
-			// Add service user role
-			if (sess.getPerunPrincipal().getUser().isServiceUser()) {
-				sess.getPerunPrincipal().getRoles().putAuthzRole(Role.SERVICEUSER);
-			}
-		}
-		sess.getPerunPrincipal().setAuthzInitialized(true);
+		refreshAuthz(sess);
```

Comment:
Please fix ``refreshAuthz(sess)`` implementation, it's not 1:1 replacement of above removed code right now.